File: leetcode/100-999/160. Intersection of Two Linked Lists/solution_3.js
/**
 * Definition for singly-linked list.
 * function ListNode(val) {
 *     this.val = val;
 *     this.next = null;
 * }
 */

 /**
  * @param {ListNode} head
  * @return {number}
  */

 function getLength(head)
 {
     let length = 0
 
     while (head)
     {
         head = head.next
         length++
     }
 
     return length
 }
 
 /**
  * @param {ListNode} headA
  * @param {ListNode} headB
  * @return {ListNode}
  */
 
 var getIntersectionNode = function(headA, headB) 
 {
     let A = headA
     let B = headB
 
     let lenA = getLength(A)
     let lenB = getLength(B)
 
     while (lenA > lenB)
     {
         A = A.next
         lenA--
     }
 
     while (lenB > lenA)
     {
         B = B.next
         lenB--
     }
 
     while (A != B)
     {
         A = A.next
         B = B.next
     }
 
     return A
 };

File: leetcode/100-999/160. Intersection of Two Linked Lists/solution_3.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) {
 *         val = x;
 *         next = null;
 *     }
 * }
 */

public class Solution 
{
    private int getLength(ListNode head)
    {
        int length = 0;

        while (head != null)
        {
            head = head.next;
            length++;
        }

        return length;
    }

    public ListNode getIntersectionNode(ListNode headA, ListNode headB) 
    {
        var A = headA;
        var B = headB;

        int lenA = getLength(A);
        int lenB = getLength(B);

        while (lenA > lenB)
        {
            A = A.next;
            lenA--;
        }

        while (lenB > lenA)
        {
            B = B.next;
            lenB--;
        }

        while (A != B)
        {
            A = A.next;
            B = B.next;
        }

        return A;
    }
}